lib/vdc/UnstructuredGrid3D.cpp

```diff
@@ -0,0 +1,278 @@
+#include <iostream>
+#include <vector>
+#include "vapor/VAssert.h"
+#include <cmath>
+#include <time.h>
+#ifdef Darwin
+    #include <mach/mach_time.h>
+#endif
+#ifdef _WINDOWS
+    #include "windows.h"
+    #include "Winbase.h"
+    #include <limits>
+#endif
+
+#include <vapor/utils.h>
+#include <vapor/vizutil.h>
+#include <vapor/UnstructuredGrid3D.h>
+
+using namespace std;
+using namespace VAPoR;
+
+UnstructuredGrid3D::UnstructuredGrid3D(const std::vector<size_t> &vertexDims, const std::vector<size_t> &faceDims, const std::vector<size_t> &edgeDims, const std::vector<size_t> &bs,
+                                       const std::vector<float *> &blks, const int *vertexOnFace, const int *faceOnVertex, const int *faceOnFace,
+                                       Location location,    // node,face, edge
+                                       size_t maxVertexPerFace, size_t maxFacePerVertex, long nodeOffset, long cellOffset, const UnstructuredGridCoordless &xug, const UnstructuredGridCoordless &yug,
+                                       const UnstructuredGridCoordless &zug, std::shared_ptr<const QuadTreeRectangle<float, size_t>> qtr)
+: UnstructuredGrid(vertexDims, faceDims, edgeDims, bs, blks, 3, vertexOnFace, faceOnVertex, faceOnFace, location, maxVertexPerFace, maxFacePerVertex, nodeOffset, cellOffset), _xug(xug), _yug(yug),
+  _zug(zug)
+{
+    VAssert(xug.GetDimensions().size() == 1);
+    VAssert(yug.GetDimensions().size() == 1);
+    VAssert(zug.GetDimensions().size() == 1);
+
+    VAssert(location == NODE);
+}
+
+
+vector<size_t> UnstructuredGrid3D::GetCoordDimensions(size_t dim) const
+{
+    if (dim == 0) {
+        return _xug.GetDimensions();
```

Comment:
These should all be the same size, correct? Just a sanity check.

Reply:
Correct